tfjs-converter/src/executor/model_analysis.ts

```diff
@@ -101,96 +101,186 @@ export function getExecutionSubgraph(
  * need to be executed to compute the output.
  */
 export function getNodesInTopologicalOrder(
-    graph: Graph, weightMap: NamedTensorsMap,
-    executionInfo: ExecutionInfo): Node[] {
+    graph: Graph, executionInfo: ExecutionInfo): Node[] {
   const {usedNodes, inputs} = executionInfo;
-  const frontier: Node[] = [];
   const inputNodes = Object.keys(inputs)
                          .map(name => parseNodeName(name)[0])
                          .map(name => graph.nodes[name]);
-  const initNodes = graph.initNodes;
+  const initNodes = graph.initNodes || [];
 
-  inputNodes.forEach(input => {
-    if (usedNodes.has(input.name)) {
-      frontier.push(input);
-    }
-  });
-  graph.weights.forEach(weight => {
-    if (usedNodes.has(weight.name)) {
-      frontier.push(weight);
-    }
-  });
-  if (initNodes != null) {
-    initNodes.forEach(node => {
-      if (usedNodes.has(node.name)) {
-        frontier.push(node);
+  const isUsed = (node: Node|string) =>
+      usedNodes.has(typeof node === 'string' ? node : node.name);
+
+  function unique(nodes: Node[]): Node[] {
+    return [...new Map(nodes.map((node) => [node.name, node])).values()];
```

Comment:
I think yes, but I won't do that because 1) it is not an contract that node objects with the same name in different map/array are always equal in reference. 2) The project-wide convention in tfjs-converter is to lookup nodes by name string instead of object reference, and I don't want to make it wrong by breaking the convention.